### PR TITLE
Add variants of SplayTree funcs without Eq Tree α

### PR DESCRIPTION
**Pull request by lorenzleutgeb**

### CODE REVIEW

File: SplayTree.ml [hunk 1]
```diff
@@ -65,34 +110,67 @@ splay_max t = match t with
...
+delete_noeq a t = match t with
+  | leaf         → leaf
+  | (tl, tx, tr) → match splay_noeq a (tl, tx, tr) with
+    | leaf       → leaf (* TODO: undefined *)
+    | (l, a', r) → if a == a'
+      then match l with
+        | leaf         → r
+        | (ll, lx, lr) → match splay_max_noeq (ll, lx, lr) with
+          | leaf          → leaf (* TODO: undefined *)
+          | (ll', m, lr') → (ll', m, lr)
```
Comment (djfo) [inline SplayTree.ml]: I think it should be `(ll', m, r)` on the right-hand side here.

**Review by djfo — CHANGES_REQUESTED** [review]
I think there's one small error, please check the comment.

File: SplayTree.ml (hunk 1 above)
Comment (lorenzleutgeb) [reply SplayTree.ml]: Nice catch, thank you!

Comment (lorenzleutgeb) [reply SplayTree.ml]: I believe that this was resolved by 1b8eecd.

Comment (djfo) [reply SplayTree.ml]: you're right

**Pull request merged**